code/modules/flufftext/Hallucination.dm

```diff
@@ -321,6 +404,45 @@ Gunshots/explosions/opening doors/less rare audio (done)
 	for(var/image/I in delusions)
 		if(target.client)
 			target.client.images.Remove(I)
+	return ..()
+
+/obj/effect/hallucination/self_delusion
+	var/image/delusion
+
+/obj/effect/hallucination/self_delusion/New(loc,mob/living/carbon/T,force_kind = null , duration = 300, custom_icon = null, custom_icon_file = null)
+	target = T
+	var/image/A = null
+	var/kind = force_kind ? force_kind : pick("clown","corgi","carp","skeleton","demon","zombie","robot")
+	switch(kind)
+		if("clown")//Clown
+			A = image('icons/mob/animal.dmi',target,"clown")
+		if("carp")//Carp
+			A = image('icons/mob/animal.dmi',target,"carp")
+		if("corgi")//Corgi
+			A = image('icons/mob/pets.dmi',target,"corgi")
+		if("skeleton")//Skeletons
+			A = image('icons/mob/human.dmi',target,"skeleton")
+		if("zombie")//Zombies
+			A = image('icons/mob/human.dmi',target,"zombie")
+		if("demon")//Demon
+			A = image('icons/mob/mob.dmi',target,"daemon")
+		if("robot")//Cyborg
+			A = image('icons/mob/robots.dmi',target,"robot")
+			target << sound('sound/voice/liveagain.ogg')
+		if("custom")
+			A = image(custom_icon_file, target, custom_icon)
+	A.override = 1
+	if(target.client)
+		target << "<span class='userdanger'>[target] is hit by a bolt of change in the chest!</span>"
```

Comment:
I fear someone will use this as an opportunity to griff, thinking they're really changed.
Plus it may be creepier to just be walking along and suddenly notice you're <i>wrong</i>.